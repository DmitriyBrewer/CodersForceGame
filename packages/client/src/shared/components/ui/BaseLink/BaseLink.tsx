import { Link as MuiLink, LinkProps as MuiLinkProps } from '@mui/material'
import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom'
import { FC } from 'react'

const BaseLink: FC<MuiLinkProps & RouterLinkProps> = props => {
  return <MuiLink component={RouterLink} {...props} />
}

export default BaseLink
