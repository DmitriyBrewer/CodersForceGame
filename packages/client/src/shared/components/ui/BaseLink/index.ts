import BaseLink from '@/shared/components/ui/BaseLink/BaseLink'

export default BaseLink
