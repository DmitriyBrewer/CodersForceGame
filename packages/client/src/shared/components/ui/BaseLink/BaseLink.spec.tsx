import { render, screen } from '@testing-library/react'

import '@testing-library/jest-dom'
import BaseLink from '@/shared/components/ui/BaseLink'

describe('<BaseLink />', () => {
  it('Snapshot is correct', () => {
    const { asFragment } = render(<BaseLink to="/" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Renders correctly with default props', () => {
    render(<BaseLink to="/" />)
    const button = screen.getByRole('a')

    expect(button).toMatchSnapshot()
  })
})
