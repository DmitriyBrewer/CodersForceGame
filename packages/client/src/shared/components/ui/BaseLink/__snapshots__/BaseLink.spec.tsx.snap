// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLink /> Renders correctly with default props 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
  href="/"
/>
`;

exports[`<BaseLink /> Snapshot is correct 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
    href="/"
  />
</DocumentFragment>
`;
